src/reconcile/rec_write.c

```diff
@@ -3467,10 +3478,19 @@ __rec_write_err(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT_PAGE *page)
      * On error, discard blocks we've written, they're unreferenced by the tree. This is not a
      * question of correctness, we're avoiding block leaks.
      */
-    for (multi = r->multi, i = 0; i < r->multi_next; ++multi, ++i)
-        if (multi->addr.block_cookie != NULL)
-            WT_TRET(__wt_btree_block_free(
-              session, multi->addr.block_cookie, multi->addr.block_cookie_size, false));
+    for (multi = r->multi, i = 0; i < r->multi_next; ++multi, ++i) {
+        if (multi->addr.block_cookie != NULL) {
+            int ret_tmp = __wt_btree_block_free(
+              session, multi->addr.block_cookie, multi->addr.block_cookie_size, r->multi_next == 1);
+            if (ret_tmp != 0 && multi->block_meta != NULL)
+                __wt_verbose_error(session, WT_VERB_WRITE,
```

Comment:
Should this be WT_VERB_WRITE or WT_VERB_BLOCK @etienneptl ?

Reply:
I would not be against sticking to the component `WT_VERB_RECONCILE`, what do you think?

Reply:
And I think we should have an error log even if `multi->block_meta` is NULL? We should just change the message.